### PR TITLE
front: inject front-end settings after build

### LINKED ISSUE #2492: Inject front-end settings after build
We need to be able to change the environment variables after the build is done.
Currently, `yarn build` will inject the variables in the final bundle, and it’s no longer possible to change them after the build is done.

### PR DESCRIPTION
**Pull request by anisometropie**
- Removed sh scripts, config has been moved to .env files. start using :
  - yarn start
  - yarn start-osrd-dev
- congured `react-inject-env`

`react-inject-env` will read from .env file, but can be overriden with shell variables

closes #2492

### CODE REVIEW

File: front/launch_dev_front.sh [hunk 1]
```diff
@@ -1,4 +1,5 @@
 #!/bin/sh
+export REACT_APP_LOCAL_BACKEND=false
 export REACT_APP_API_URL=https://gateway.dev.dgexsol.fr/osrd
 export REACT_APP_CHARTIS_URL=https://gateway.dev.dgexsol.fr/chartos
 export REACT_APP_EDITOAST_URL=https://gateway.dev.dgexsol.fr/editoast
```
Comment (flomonster) [inline front/launch_dev_front.sh]: This file is not used and should be deleted

**Review by flomonster — CHANGES_REQUESTED** [review]
Great we should also update the docker compose files

**Comment by alexandredamiron** [comment]
front/README.md should be updated, save interns soul

**Review by alexandredamiron — CHANGES_REQUESTED** [review]
README.md must be updated, save interns soul

File: docker-compose.yml [hunk 2]
```diff
@@ -113,10 +113,10 @@ services:
...
       args:
         NGINX_CONFIG: "nginx-dev.conf"
-        REACT_APP_LOCAL_BACKEND: "True"
-        REACT_APP_API_URL: "http://localhost:8000"
-        REACT_APP_CHARTIS_URL: "http://localhost:7070"
-        REACT_APP_EDITOAST_URL: "http://localhost:8090"
+        REACT_APP_LOCAL_BACKEND: "${REACT_APP_LOCAL_BACKEND:-True}"
+        REACT_APP_API_URL: "${REACT_APP_API_URL:-http://localhost:8000}"
+        REACT_APP_CHARTIS_URL: "${REACT_APP_CHARTIS_URL:-http://localhost:7070}"
+        REACT_APP_EDITOAST_URL: "${REACT_APP_EDITOAST_URL:-http://localhost:8090}"
```
Comment (flomonster) [inline docker-compose.yml]: Few things here:
- These variable must be moved to environment
- It's not really useful to get the use var env if it exists
- We can rename them (removing the prefix `REACT_APP_`)

**Review by flomonster — CHANGES_REQUESTED** [review]

**Review by flomonster — APPROVED** [review]

**Review by alexandredamiron — APPROVED** [review]

**Pull request merged**